diff --git a/boards/syna/sr100/board.yml b/boards/syna/sr100/board.yml
deleted file mode 100644
index 6758b0e02fc..00000000000
--- a/boards/syna/sr100/board.yml
+++ /dev/null
@@ -1,6 +0,0 @@
-boards:
-  - name: sr100_rdk
-    full_name: SR100 RDK
-    vendor: syna
-    socs:
-      - name: sr100
diff --git a/drivers/clock_control/clock_control_syna_sr100.c b/drivers/clock_control/clock_control_syna_sr100.c
index b8fffe8e19d..aa9a19d2960 100644
--- a/drivers/clock_control/clock_control_syna_sr100.c
+++ b/drivers/clock_control/clock_control_syna_sr100.c
@@ -11,7 +11,6 @@
 
 #define DT_DRV_COMPAT syna_sr100_clock
 
-#define PLL0_RATE CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC
 #define CLK_ENABLE1 0x0
 
 #define DIV_EN    (1 << 2)
@@ -29,6 +28,7 @@
 
 struct clock_control_syna_config {
 	mem_addr_t regs;
+	uint32_t perif_rate;
 };
 
 static inline uint32_t syna_clk_read(const struct clock_control_syna_config *config, mm_reg_t addr)
@@ -45,7 +45,7 @@ static inline void syna_clk_write(const struct clock_control_syna_config *config
 static int syna_clk_set_rate(const struct device *dev, uint32_t id, uint32_t rate)
 {
 	const struct clock_control_syna_config *config = dev->config;
-	uint32_t parent_rate = PLL0_RATE;
+	uint32_t parent_rate = config->perif_rate;
 	uint32_t offset = GET_CGL_REG(id);
 	uint32_t divider = 0;
 	uint32_t value;
@@ -84,7 +84,7 @@ static uint32_t syna_clk_get_rate(const struct device *dev, uint32_t id)
 {
 	const struct clock_control_syna_config *config = dev->config;
 	uint32_t offset = GET_CGL_REG(id);
-	uint32_t parent_rate = PLL0_RATE;
+	uint32_t parent_rate = config->perif_rate;
 	uint32_t divider = 1;
 	uint32_t value;
 
@@ -221,6 +221,7 @@ static DEVICE_API(clock_control, syna_clkctrl_api) = {
 
 static const struct clock_control_syna_config syna_config = {
 	.regs = DT_INST_REG_ADDR(0),
+	.perif_rate = DT_INST_PROP_OR(0, perif_rate, CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC),
 };
 
 static int syna_clkctrl_init(const struct device *dev)
diff --git a/dts/bindings/clock/syna,sr100-clock.yaml b/dts/bindings/clock/syna,sr100-clock.yaml
index 1c55748f1e1..5e0be740e6a 100644
--- a/dts/bindings/clock/syna,sr100-clock.yaml
+++ b/dts/bindings/clock/syna,sr100-clock.yaml
@@ -14,5 +14,10 @@ properties:
   "#clock-cells":
     const: 1
 
+  perif-rate:
+    type: int
+    description: |
+      Clock rate of peripherals
+
 clock-cells:
   - clkid
diff --git a/soc/syna/sr100/Kconfig b/soc/syna/sr100/Kconfig
deleted file mode 100644
index 4c867011452..00000000000
--- a/soc/syna/sr100/Kconfig
+++ /dev/null
@@ -1,24 +0,0 @@
-# Copyright (c) 2025 Synaptics, Inc.
-# SPDX-License-Identifier: Apache-2.0
-
-config SOC_SR100
-	select ARM
-	select ARM_MPU
-	select CLOCK_CONTROL
-	select CPU_CORTEX_M55
-	select DYNAMIC_INTERRUPTS
-	select CPU_CORTEX_M_HAS_SYSTICK
-	select CPU_CORTEX_M_HAS_DWT
-	select CPU_HAS_ARM_MPU
-	select CPU_HAS_ARM_SAU
-	select CPU_HAS_DCACHE
-	select CPU_HAS_ICACHE
-	select CPU_HAS_FPU
-	select UART_NS16550_ACCESS_WORD_ONLY if UART_NS16550
-	select SERIAL_SUPPORT_ASYNC
-	select HAS_CMSIS_CORE
-	select SOC_EARLY_INIT_HOOK
-	select RESET
-
-config SOC_PART_NUMBER
-	default "SR100" if SOC_SR100
diff --git a/soc/syna/sr100/soc.yml b/soc/syna/sr100/soc.yml
deleted file mode 100644
index 01428376f2f..00000000000
--- a/soc/syna/sr100/soc.yml
+++ /dev/null
@@ -1,4 +0,0 @@
-series:
-- name: sr1xx
-  socs:
-  - name: sr100
