diff --git a/drivers/serial/Kconfig.ns16550 b/drivers/serial/Kconfig.ns16550
index 3a1d35eda16..b7ba0441762 100644
--- a/drivers/serial/Kconfig.ns16550
+++ b/drivers/serial/Kconfig.ns16550
@@ -31,6 +31,13 @@ config UART_NS16550_DRV_CMD
 
 	  Says n if not sure.
 
+config UART_NS16550_SYNA_DMA
+	bool "Synaptics DMA support for NS16550"
+	select SERIAL_SUPPORT_ASYNC
+	select DMA if UART_ASYNC_API
+	help
+	  This enables the usage of Synaptics internal DMA for Async operations.
+
 config UART_NS16550_INTEL_LPSS_DMA
 	bool "INTEL LPSS support for NS16550"
 	select SERIAL_SUPPORT_ASYNC
