diff --git a/drivers/ethernet/eth_dwmac.c b/drivers/ethernet/eth_dwmac.c
index a6cd517a680..715efd2850c 100644
--- a/drivers/ethernet/eth_dwmac.c
+++ b/drivers/ethernet/eth_dwmac.c
@@ -85,13 +85,13 @@ static inline uint32_t lo32(uintptr_t val)
 	return val;
 }
 
-static inline uint32_t phys_hi32(void *addr)
+__weak uint32_t phys_hi32(void *addr)
 {
 	/* the default 1:1 mapping is assumed */
 	return hi32((uintptr_t)addr);
 }
 
-static inline uint32_t phys_lo32(void *addr)
+__weak uint32_t phys_lo32(void *addr)
 {
 	/* the default 1:1 mapping is assumed */
 	return lo32((uintptr_t)addr);
@@ -130,7 +130,7 @@ static inline int net_pkt_get_nbfrags(struct net_pkt *pkt)
 static int dwmac_send(const struct device *dev, struct net_pkt *pkt)
 {
 	struct dwmac_priv *p = dev->data;
-	struct net_buf *frag, *pinned;
+	struct net_buf *frag;
 	unsigned int pkt_len = net_pkt_get_len(pkt);
 	unsigned int d_idx;
 	struct dwmac_dma_desc *d;
@@ -156,17 +156,12 @@ static int dwmac_send(const struct device *dev, struct net_pkt *pkt)
 			goto abort;
 		}
 
-		/* pin this fragment */
-		pinned = net_buf_clone(frag, TX_AVAIL_WAIT);
-		if (!pinned) {
-			LOG_DBG("net_buf_clone() returned NULL");
-			k_sem_give(&p->free_tx_descs);
-			goto abort;
-		}
-		sys_cache_data_flush_range(pinned->data, pinned->len);
-		p->tx_frags[d_idx] = pinned;
-		LOG_DBG("d[%d]: frag %p pinned %p len %d", d_idx,
-			(void *)frag->data, (void *)pinned->data, pinned->len);
+		/* Take reference for the DMA */
+		net_pkt_frag_ref(frag);
+
+		sys_cache_data_flush_range(frag->data, frag->len);
+		p->tx_frags[d_idx] = frag;
+		LOG_DBG("d[%d]: frag %p len %d", d_idx, (void *)frag->data, frag->len);
 
 		/* if no more fragments after this one: */
 		if (!frag->frags) {
@@ -177,9 +172,9 @@ static int dwmac_send(const struct device *dev, struct net_pkt *pkt)
 
 		/* fill the descriptor */
 		d = &p->tx_descs[d_idx];
-		d->des0 = phys_lo32(pinned->data);
-		d->des1 = phys_hi32(pinned->data);
-		d->des2 = pinned->len | des2_flags;
+		d->des0 = phys_lo32(frag->data);
+		d->des1 = phys_hi32(frag->data);
+		d->des2 = frag->len | des2_flags;
 		d->des3 = pkt_len | des3_flags;
 
 		/* clear the FD flag on subsequent descriptors */
@@ -674,7 +669,8 @@ int dwmac_probe(const struct device *dev)
 	memset(p->rx_descs, 0, NB_RX_DESCS * sizeof(struct dwmac_dma_desc));
 
 	/* set up DMA */
-	REG_WRITE(DMA_CHn_TX_CTRL(0), 0);
+	REG_WRITE(DMA_CHn_TX_CTRL(0),
+		  FIELD_PREP(DMA_CHn_TX_CTRL_PBL, 32));
 	REG_WRITE(DMA_CHn_RX_CTRL(0),
 		  FIELD_PREP(DMA_CHn_RX_CTRL_PBL, 32) |
 		  FIELD_PREP(DMA_CHn_RX_CTRL_RBSZ, RX_FRAG_SIZE));
